core/templates/dev/head/i18n.js

```diff
@@ -45,20 +45,26 @@ oppia.constant('DEFAULT_TRANSLATIONS', {
 });
 
 oppia.controller('I18nFooter', [
-    '$rootScope', '$scope', '$translate',
-    function($rootScope, $scope, $translate) {
+    '$http', '$rootScope', '$scope', '$translate',
+    function($http, $rootScope, $scope, $translate) {
   // Changes the language of the translations.
+  var _PREFERENCES_DATA_URL = '/preferenceshandler/data';
+  var _SAVE_SITE_LANGUAGE_URL = '/save_site_language';
   $scope.supportedSiteLanguages = GLOBALS.SUPPORTED_SITE_LANGUAGES;
+  if (GLOBALS.userIsLoggedIn) {
+    $http.get(_PREFERENCES_DATA_URL).success(function(data) {
+      $translate.use(data.preferred_site_language_code);
+    });
+  }
   $scope.changeLanguage = function(langCode) {
     $translate.use(langCode);
+    if (GLOBALS.userIsLoggedIn) {
+      $http.put(_SAVE_SITE_LANGUAGE_URL, {
+        site_language_code: langCode,
+        requestIsFromFooter: true
```

Comment:
Can the information that the request is from the footer be an additional argument to put(), rather than being bundled in with the rest of the data? From the angular docs it looks like .put() supports an additional config property; maybe this can go there.